tests/store/artifact/test_databricks_artifact_repo.py

```diff
@@ -591,6 +556,226 @@ def test_paginated_list_artifacts(self, databricks_artifact_repo):
+            message_mock.assert_has_calls(
+                [
+                    mock.call(GetCredentialsForRead(run_id=MOCK_RUN_ID, path=["testpath"])),
+                    mock.call(
+                        GetCredentialsForRead(run_id=MOCK_RUN_ID, path=["testpath"], page_token="2")
+                    ),
+                    mock.call(
+                        GetCredentialsForRead(run_id=MOCK_RUN_ID, path=["testpath"], page_token="3")
+                    ),
+                ]
+            )
+            assert call_endpoint_mock.call_count == 3
+
+    def test_get_read_credential_infos_respects_max_request_size(self, databricks_artifact_repo):
+        """
+        Verifies that the `_get_read_credential_infos` method, which is used to resolve read access
+        credentials for a collection of artifacts, handles paginated responses properly, issuing
+        incremental requests until all pages have been consumed
+        """
+        # Create 3 chunks of paths, two of which have the maximum request size and one of which
+        # is smaller than the maximum chunk size. Aggregate and pass these to
+        # `_get_read_credential_infos`, validating that this method decomposes the aggregate
+        # list into these expected chunks and makes 3 separate requests
+        paths_chunk_1 = ["path1"] * 2000
+        paths_chunk_2 = ["path2"] * 2000
+        paths_chunk_3 = ["path3"] * 5
+        credential_infos_mock_1 = [
+            ArtifactCredentialInfo(
+                signed_uri="http://mock_url_1", type=ArtifactCredentialType.AWS_PRESIGNED_URL
+            )
+            for _ in range(2000)
+        ]
+        credential_infos_mock_2 = [
+            ArtifactCredentialInfo(
+                signed_uri="http://mock_url_2", type=ArtifactCredentialType.AWS_PRESIGNED_URL
+            )
+            for _ in range(2000)
+        ]
+        credential_infos_mock_3 = [
+            ArtifactCredentialInfo(
+                signed_uri="http://mock_url_3", type=ArtifactCredentialType.AWS_PRESIGNED_URL
+            )
+            for _ in range(5)
+        ]
+
+        with mock.patch(
+            DATABRICKS_ARTIFACT_REPOSITORY_PACKAGE + ".message_to_json"
+        ) as message_mock, mock.patch(
+            DATABRICKS_ARTIFACT_REPOSITORY + "._call_endpoint"
+        ) as call_endpoint_mock:
+            call_endpoint_mock.side_effect = [
+                GetCredentialsForRead.Response(credential_infos=credential_infos_mock_1,),
+                GetCredentialsForRead.Response(credential_infos=credential_infos_mock_2,),
+                GetCredentialsForRead.Response(credential_infos=credential_infos_mock_3,),
+            ]
+
+            databricks_artifact_repo._get_read_credential_infos(
+                MOCK_RUN_ID, paths_chunk_1 + paths_chunk_2 + paths_chunk_3,
+            )
+            assert call_endpoint_mock.call_count == message_mock.call_count == 3
```

Comment:
asserting the 2 conditions separately in case one fails the other doesn't.
assert a == 3
assert b == 3

Reply:
Good point! Done!